src/store/Controller.cc

```diff
@@ -355,6 +355,13 @@ Store::Controller::allowSharing(StoreEntry &entry, const cache_key *key)
         const bool found = anchorToCache(entry, inSync);
         if (found && !inSync)
             throw TexcHere("cannot sync");
+        if (!found) {
+            assert(transients->hasWriter(entry));
```

Comment:
I hesitated when deciding whether we should assert here or report to admin. However, since we now remove idle Transients entries, there should be no situation when both '!found' and !hasWriter() conditions are true: if the writer disappears before delivering the response, it should 'mark' the entry (making it unavailable). Similarly, for a cached response without writer, 'found' should be true.

Reply:
We cannot assert that `hasWriter` _stays_ true _after_ a false `found` was computed.

Reply:
Fixed at a9cfe77.